sdk/core/Azure.Core/src/ResourceType.cs

```diff
@@ -22,47 +28,28 @@ public ResourceType(string resourceType)
         {
             Argument.AssertNotNullOrWhiteSpace(resourceType, nameof(resourceType));
 
-            int index = resourceType.IndexOf(ResourceIdentifier.Separator);
-            if (index == -1 || resourceType.Length <3)
+            _namespaceSeparatorIndex = resourceType.IndexOf(ResourceIdentifier.Separator);
+            if (_namespaceSeparatorIndex == -1 || resourceType.Length < 3)
                 throw new ArgumentOutOfRangeException(nameof(resourceType));
 
             _stringValue = resourceType;
-            Namespace = resourceType.Substring(0, index);
-            Type = resourceType.Substring(index + 1);
-        }
-
-        /// <summary>
-        /// Create a resource type given the namespace and typename components.
-        /// </summary>
-        /// <param name="providerNamespace"></param>
-        /// <param name="name"></param>
-        internal ResourceType(string providerNamespace, string name)
-        {
-            Namespace = providerNamespace;
-            Type = name;
-            _stringValue = $"{Namespace}{ResourceIdentifier.Separator}{Type}";
-        }
-
-        internal ResourceType AppendChild(string childType)
-        {
-            return new ResourceType($"{_stringValue}{ResourceIdentifier.Separator}{childType}");
         }
 
         /// <summary>
         /// Gets the last resource type name.
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public string GetLastType() => Type.Substring(Type.LastIndexOf(ResourceIdentifier.Separator) + 1);
+        public string GetLastType() => _stringValue.Substring(_stringValue.LastIndexOf(ResourceIdentifier.Separator) + 1);
```

Comment:
nit: could cache the last index as well.

Reply:
I believe caching that would require us to make the struct non readonly which we are trying to not do.